src/server/generic_family.cc

```diff
@@ -1514,28 +1517,29 @@ OpStatus GenericFamily::OpMove(const OpArgs& op_args, string_view key, DbIndex t
   auto& db_slice = op_args.shard->db_slice();
 
   // Fetch value at key in current db.
-  auto [from_it, from_expire] = db_slice.FindExt(op_args.db_cntx, key);
-  if (!IsValid(from_it))
+  auto from_res = db_slice.FindMutable(op_args.db_cntx, key);
+  if (!IsValid(from_res.it))
     return OpStatus::KEY_NOTFOUND;
 
   // Fetch value at key in target db.
   DbContext target_cntx = op_args.db_cntx;
   target_cntx.db_index = target_db;
-  auto [to_it, _] = db_slice.FindExt(target_cntx, key);
-  if (IsValid(to_it))
+  auto to_res = db_slice.FindReadOnly(target_cntx, key);
+  if (IsValid(to_res.it))
     return OpStatus::KEY_EXISTS;
 
   // Ensure target database exists.
   db_slice.ActivateDb(target_db);
 
-  bool sticky = from_it->first.IsSticky();
-  uint64_t exp_ts = db_slice.ExpireTime(from_expire);
-  PrimeValue from_obj = std::move(from_it->second);
+  bool sticky = from_res.it->first.IsSticky();
+  uint64_t exp_ts = db_slice.ExpireTime(from_res.exp_it);
+  PrimeValue from_obj = std::move(from_res.it->second);
 
   // Restore expire flag after std::move.
-  from_it->second.SetExpire(IsValid(from_expire));
+  from_res.it->second.SetExpire(IsValid(from_res.exp_it));
 
-  CHECK(db_slice.Del(op_args.db_cntx.db_index, from_it));
+  from_res.post_updater.Run();
```

Comment:
In line 1536 we call std::move on the value. I feel we have a bug here as we restore only the expire flag after that. What if we have other flags turned on such as external entry. What will update in post update run? will it count the correct values?

Reply:
This is an existing bug, right?
I'm not sure how `PostUpdate()` will handle it, but it might be a good idea to load from memory both src and dest. Is this not part of the flow anyway?

Reply:
This is an existing bug.
1. We can run the post update on from before we are doting the std::move we are not going to update it but delete it next
2. I am not sure how the db_slice.Del works after we call the std::move , we can create a github issue for this and add there to add a check that we are not calling move on io pending /external entry as this is not supported in this flow.

Reply:
Moved to before the `move()`.
What's wrong with `Del()` after `move()`?